libethash-cuda/CUDAMiner.cpp

```diff
@@ -135,8 +134,9 @@ void CUDAMiner::workLoop()
 
 void CUDAMiner::kick_miner()
 {
-	if (!m_abort)
-		m_abort = true;
+	bool f = false;
+	// weak if ok here. If it fails it's because it was already true!
+	m_abort.compare_exchange_weak(f, true);
```

Comment:
It looks only the pool thread is setting this to `true`. Also, according to the comment, we can ignore the case where it is already `true`. Then, just use `m_abort.store(true)` here. To get better performance, you can use `m_abort.store(true, std::memory_order_relaxed`.

Reply:
Darn. Of course.